.github/workflows/linter.yml

```diff
@@ -0,0 +1,16 @@
+name: Linter
+
+on:
+  pull_request:
```

Comment:
I know that you originally wanted to run the action only on PRs against the master branch, but I highly suggest running it for all PRs.

You can tweak and optimize later, if needed.